### ISSUE TITLE
StrikethroughExtension does not work with links

### ISSUE THREAD

**Issue by adam-boduch** [thephpleague/commonmark#867]
### Version(s) affected

2.3.0

### Description

StrikethroughExtension does not work with links. 

Input: `~~Prefix i link: https://aws.amazon.com/emr/features/hadoop/~~`
Expected output: 

```
<p>
  <del>
    Prefiks i link: <a href="https://aws.amazon.com/emr/features/hadoop/">https://aws.amazon.com/emr/features/hadoop/
    </a>
  </del>
</p
```

CommonMark output: 
```
<p>
  ~~Prefiks i link: https://aws.amazon.com/emr/features/hadoop/~~
</p>
```

### How to reproduce

```php
$environment = new Environment([]);
$environment->addExtension(new CommonMarkCoreExtension());
$environment->addExtension(new StrikethroughExtension());
$environment->addExtension(new AutolinkExtension());

$converter = new MarkdownConverter($environment);
$document = $converter->convert($text);

return (string) $document;
```

**Comment by colinodell** [comment]
Yep, looks like we're not parsing this the same way as GFM: https://babelmark.github.io/?text=~~Prefix+i+link%3A+https%3A%2F%2Faws.amazon.com%2Femr%2Ffeatures%2Fhadoop%2F~~  Thanks for the report!

**Comment by colinodell** [comment]
Interestingly, it does work correctly when you wrap the URL in `<` and `>`:

```md
~~Prefix i link: <https://aws.amazon.com/emr/features/hadoop/>~~
```

So maybe AutolinkExtension is the culprit here?  I'll have to dig into this further.

**Comment by colinodell** [comment]
This has been fixed in versions 2.3.1 and 2.2.4.  Thanks again for reporting this!
